### ISSUE TITLE
RFC: Improving ClipSpace performance

### ISSUE THREAD

**Issue by mattes3** [uber/luma.gl#1203]
#### Background

luma's [ClipSpace](https://github.com/uber/luma.gl/blob/72d3e96ab960915865f7d0dcd58f6a5ab64aa23b/modules/core/src/lib/clip-space.js) currently works with 2 triangles that form a square. Other frameworks (e.g. gl-react) use one big triangle, instead. There even is an npm component called [a-big-triangle](https://github.com/mikolalysenko/a-big-triangle) that shows this idea. It points to [a blog post](https://michaldrobot.com/2014/04/01/gcn-execution-patterns-in-full-screen-passes/) that claims that one triangle is about 8% faster than 2 triangles and leads better results.

I'm NOT an expert who could judge this difference or who could change the code correctly, therefore, a question to y'all:

* Do you find it worthwhile to replace the 2 triangles by 1? Maybe it would make luma.gl faster.

#### To Do List
  - [x] Add label and assign to milestone
  - [x] Coding
  - [x] Doc update
  - [x] Whats new update
  - [x] Test

**Comment by tsherif** [comment]
This is quite interesting! Thank you for the references. It should be easy to run benchmarks on a few different hardware setups to see if it's worth making the change.

**Comment by ibgreen** [comment]
Yes, could be worth a look. I seem to remember considering using a single larger triangle, but opted for using two perfectly fitted triangles instead. No strong reasons I believe, I was mainly focused on getting things functionally correct, didn't do any benchmarks.
